crates/proof-of-sql/src/base/slice_ops/inner_product.rs

```diff
@@ -26,3 +27,11 @@ where
         .map(|(a, b)| a.into() * *b)
         .sum()
 }
+
+/// Cannot use blanket impls for `Vec<u8>` because bytes might have different embeddings as scalars
+pub fn inner_product_with_bytes<S: ScalarExt>(a: &[Vec<u8>], b: &[S]) -> S {
```

Comment:
It would be slightly nicer to be able to pass a function in for the conversion so you could do something like

```rust
inner_product_with(a, b, ScalarExt::from_byte_slice_via_hash)
```
that way, you can _probably_ make `inner_product_ref_cast` just be
```rust
inner_product_with(a, b, Into::into)
```
or something similar, resulting in less duplicate code.

Reply:
Also, there are no tests for this.